src/main/java/com/github/javabdd/JFactory.java

```diff
@@ -2643,6 +2657,260 @@ int relnext_rec(int states, int relation, int vars) {
+                }
+                result = relnextUnion_rec(states, relation, union, vars);
+
+                if (numReorder == 0) {
+                    bdd_enable_reorder();
+                }
+            } catch (ReorderException x) {
+                bdd_checkreorder();
+                numReorder--;
+                continue again;
+            }
+            break;
+        }
+
+        checkresize();
+        return result;
+    }
+
+    int relnextUnion_rec(int states, int relation, int union, int vars) {
+        if (VERIFY_ASSERTIONS) {
+            _assert(!ZDD);
+        }
+
+        if (cachestats.enabled) {
+            cachestats.opAccess++;
+        }
+
+        // Terminals cases.
+        if (ISZERO(union)) {
+            return relnext_rec(states, relation, vars);
+        }
+        if (ISONE(union)) {
+            return bddtrue;
+        }
+        if (ISZERO(states) || ISZERO(relation)) {
+            return union;
+        }
+        if (ISONE(states) && ISONE(relation)) {
+            return bddtrue;
+        }
+        if (ISCONST(vars)) {
+            return or_rec(states, union);
+        }
+
+        // Determine the top level of 'states' and 'relation'.
+        int level_states = LEVEL(states);
+        int level_relation = LEVEL(relation);
+        int level = level_states < level_relation ? level_states : level_relation;
+
+        // Skip levels in the variable set until we reach (or exceed) 'level'.
+        boolean sameHeight = false;
+
+        for (;;) {
+            int level_vars = LEVEL(vars);
+
+            if (level == level_vars || (level ^ 1) == level_vars) {
+                sameHeight = true;
+                break;
+            }
+
+            if (level < level_vars) {
+                break;
+            }
+
+            vars = HIGH(vars);
+
+            if (ISCONST(vars)) {
+                return or_rec(states, union);
+            }
+        }
+
+        // Consult the operation cache.
+        BddCacheDataI entry = BddCache_lookupI(itecache, QUINTUPLE(states, relation, union, vars, bddop_relnextUnion));
+
+        if (entry.a == states && entry.b == relation && entry.c == union && entry.d == vars
+                && entry.e == bddop_relnextUnion)
+        {
+            if (cachestats.enabled) {
+                cachestats.opHit++;
+            }
+            return entry.res;
+        }
+
+        if (cachestats.enabled) {
+            cachestats.opMiss++;
+        }
+
+        // Could not find a cached result, so perform the recursive operation to compute the result.
+        int result;
+
+        int level_union = LEVEL(union);
+
+        if (level_union < (level & (~1))) {
+            PUSHREF(relnextUnion_rec(states, relation, LOW(union), vars));
+            PUSHREF(relnextUnion_rec(states, relation, HIGH(union), vars));
+            result = bdd_makenode(level_union, READREF(2), READREF(1));
+            POPREF(2);
+        } else {
+            if (sameHeight) {
```

Comment:
`relprevUnion` and `relnextUnion` differ in `} else if (sameHeight) {` vs `} else {\nif (sameHeight) {`. Is there a reason for that?

Reply:
No, I overlooked that. I now changed this into 'else if' for consistency. (The diff seems quite large though for such a simple change.)